### PR TITLE
Map update and remove pipeline functions

### LINKED ISSUE #6470: Modify map in pipelines
## Expected Behavior

I'd like to have a way to modify maps in pipelines. Particularly, I'm looking for ways to do one of:

1. remove a key from a map
2. assign a value to a map key

I'm looking for something that would allow me to do something equivalent to:

```
// assign/replace
map["timestamp"] = parse_date(map["timestamp"]);
// remove
remove_key(map, "timestamp");
```

## Current Behavior
There's no way to modify map variables after they're created (that I could find in the documentation)

## Context
I'm trying to convert a `timestamp` field present in some log groups messages to an actual date object in a pipeline. See https://community.graylog.org/t/removing-key-from-pipeline-map/12037

## Your Environment

* Graylog Version: 3.1.0
* Elasticsearch Version: AWS 6.8
* MongoDB Version: 4.2.0

### PR DESCRIPTION
**Pull request by patrickmann**
Resolves #6470 

Introduces new functions `map_set` and `map_remove`.

Here is a sample rule for testing:

> rule "t1"
> when
>   true
> then
>    let map = {k1: "v1", k2: "v2"};
>    set_field("a1",to_string(map));
>    map_remove(map,"k1");
>    set_field("a2",to_string(map));
>    map_set(map,"k1","v3");
>    map_set(map,"k3",3.4);
>    set_field("a3",to_string(map));
> end

### CODE REVIEW

**Review by bernd — COMMENTED** [review]
Two questions:

- How about naming the function `map_set` instead of `map_upsert`? This would be consistent with functions like `lookup_set_value`.
- The value seems to be of type "String". If that's the only value type that is supported, we might want to reflect that in the function name. But I think we should rather allow multiple types instead.

File: graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/functions/maps/MapSet.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+import java.util.Map;
+
+import static com.google.common.collect.ImmutableList.of;
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.object;
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.string;
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.type;
+
+public class MapSet extends AbstractFunction<Map> {
+    private static final ObjectMapper MAPPER = new ObjectMapper();
```
Comment (moesterheld) [inline graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/functions/maps/MapSet.java:36]:
```suggestion
```

remove unused `ObjectMapper`

**Review by moesterheld — APPROVED** [review]
lgtm. just some minor comments

File: graylog2-server/src/test/resources/org/graylog/plugins/pipelineprocessor/functions/mapRemove.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+rule "mapRemove"
+when
+  true
+then
+  let newValue = to_map(parse_json("{\"k1\":\"v1\",\"k2\":\"v2\"}"));
```
[the remark this answers is not in the document]
Comment (patrickmann) [reply graylog2-server/src/test/resources/org/graylog/plugins/pipelineprocessor/functions/mapRemove.txt:5]: Didn't know you could assign a map literal like that - thx.

**Review by moesterheld — APPROVED** [review]

**Pull request merged**